services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -286,6 +285,13 @@ protected boolean cleanupBackupVersion(Store store, String clusterName) {
     return true;
   }
 
+  private boolean readyToBeRemoved(Version v, boolean isRepush, int currentVersionNum) {
+    if (isRepush) {
+      return v.getCreatedTime() == Duration.ZERO.toMillis();
+    }
+    return v.getNumber() < currentVersionNum;
+  }
+
```

Comment:
I wondered whether this logic will work well or not.
Let us say we have two versions:
v1,
v2,
And v3 is the repushed version from v2.

So when cleaning up the store versions, v2 will be removed as its created time is 0, and the true backup version will never be removed as its createdTime is not `0`, right?

Reply:
It will be removed in the later iteration after regular push, right?

Reply:
Well, what if there is no regular push in the following month? 
This comment is related to the following comment, and the main cause is that we can't differentiate which version is the source version for the repushed current version, and if we have that info, the logic will be very straightforward.
The source version cleanup will be independent from the true backup version cleanup.